requirements.txt

```diff
@@ -0,0 +1,13 @@
+fsspec
```

Comment:
Unless I missed it, I don't see fsspec used anywhere so let's leave it out. If it's a dependency for something else, pip will take care of it.

Also, there is no need for the requirements.txt file since pip can install dependencies from pyproject.toml (e.g. `py -m pip install -e .[dev]`)